src/components/Profile/index.js

```diff
@@ -0,0 +1,183 @@
+            </Button>
+          </ButtonGroup>
+        )}
+        {isTeacher && teacherUpcomingSessions && teacherOrUser && (
+          <Container>
+            <br />
+            {!teacher && <h2>Sessions you teach</h2>}
+            <br />
+            <br />
+            <h2>Upcoming Sessions</h2>
+            <br />
+            <CardDeck>
+              {teacherUpcomingSessions.map((session) => (
+                <div key={session.id}>
+                  <Session
+                    session={session}
+                    teacher={false}
+                    btn={false}
+                    bg="light"
+                  />
+                  {participants && <Participants session={session} />}
+                </div>
+              ))}
+            </CardDeck>
+          </Container>
+        )}
+        {isTeacher && teacherPastSessions && teacherOrUser && (
+          <Container>
+            <br />
+            <h2>Past Sessions</h2>
+            <br />
+            <CardDeck>
+              {teacherPastSessions.map((session) => (
+                <div key={session.id}>
+                  <Session
+                    session={session}
+                    teacher={false}
+                    btn={false}
+                    bg="secondary"
+                  />
+                  {participants && <Participants session={session} />}
+                </div>
+              ))}
+            </CardDeck>
+          </Container>
+        )}
+        {!teacher && participantUpcomingSessions && attend && (
+          <Container>
+            <br />
+            <h2>Sessions you attend</h2>
+            <br />
+            <br />
+            <h2>Upcoming Sessions</h2>
+            <br />
+            <CardDeck>
+              {participantUpcomingSessions.map((session) => (
+                <Session
+                  key={session.id}
+                  session={session}
+                  teacher
+                  btn
+                  bg="light"
+                />
+              ))}
+            </CardDeck>
+          </Container>
+        )}
+        {!teacher && participantPastSessions && attend && (
+          <Container>
+            <br />
+            <h2>Past Sessions</h2>
+            <br />
+            <CardDeck>
+              {participantPastSessions.map((session) => (
+                <Session
+                  key={session.id}
+                  session={session}
+                  teacher
+                  btn={false}
+                  bg="secondary"
+                />
+              ))}
+            </CardDeck>
+          </Container>
+        )}
+        {isTeacher && profile.receivedReviews && (
+          <Container>
+            <br />
+            <h2>Reviews</h2>
+            <br />
+            {profile.receivedReviews.map((review) => (
+             review.reviewer && <Review key={review.id} review={review} />
+            ))}
+          </Container>
+        )}
+      </Container>
+    </div>
+  );
+}
```

Comment:
This component seems too big but i don't have a clear way to split it without going deep into all those conditions